pkg/rehearse/jobs.go

```diff
@@ -605,6 +605,10 @@ func pickTemplateJob(presubmits map[string][]prowconfig.Presubmit, templateFile,
 				continue
 			}
 
+			if job.Hidden {
+				continue
+			}
```

Comment:
For clarity, I'll just comment that (as opposed to the previous comments for `GetChanged*` methods) this hunk should stay - it's in our interest to pick up a job that won't be later filtered out by the `filter*` methods.

This also made me spot a probable bug, btw: `pickTemplateJob` will probably happily pick a job without an `rehearsable` label, which will later be filtered out because of that. Looks like we should extract that whole logic (in `filter` methods ) that decide whether a job is reheasable or not and also use it here to check that we won't pick an unrehearsable job, so that we avoid duplicating that code.

Reply:
Nice catch. It seems that I missed updating this part when I added the rehearse label.